features/project/project_controller.rb

```diff
@@ -70,11 +70,32 @@ class ProjectController < AuthenticatedControllerBase
       erb(:new_project, locals: locals, layout: FastlaneCI.default_layout)
     end
 
+    get "#{HOME}/lanes/*/*/*" do |org, repo_name, branch|
+      content_type :json
+
+      provider_credential = check_and_get_provider_credential(type: FastlaneCI::ProviderCredential::PROVIDER_CREDENTIAL_TYPES[:github])
+
+      selected_repo = github_service.repos.detect { |repo| repo_name == repo.name }
```

Comment:
This will also make sure you can only access files you actually have access to, right?

Reply:
`GitHubService` uses the `api_token` from the `provider_credential` so it'll only show up repos which the user has access to